### PR TITLE
Feature/proposal page data wireup

### LINKED ISSUE #542: Proposal page initial real data
Currently we have proposal data being loaded in to the main guilds page but still have purely hardcoded placeholder data in proposal view (https://develop.dxvote.dev/#/guilds/rinkeby/0x9cdc16b5f95229b856cba5f38095fd8e00f8edef/proposal/5)

We need to add a clickable link to the proposal cards on the main page that goes to ./proposal/proposalID
We will then pull the proposal ID from the URL.

We should use the existing contexts to get the data. 

Then we just need to place the actual data in place of the placeholders. This can be done initially with the:
- Title
- Description (from contentHash - use the existing ipfs code from IPFSService.ts)
- Start and end time

### PR DESCRIPTION
**Pull request by mprasanjith**
This change introduces real data from the guilds smart contracts into the proposal page. Changes to show the actual sidebar content and call data will be included in a follow-up PR.

image

Remarks:
- I didn't access the ProposalsContext to get the proposal data as we probably will refactor this soon with the `swr` implementation. 
- The WalletButton component has been split into two. The new `AddressButton` component is a generic component that supports showing any Ethereum address and avatar.
- We should define a schema for the proposal metadata stored in IPFS soon. For now I added a placeholder `ProposalMetadata` interface with just a description field.
- We need to have a generic error component to show in case something failed while loading data.

Closes #542.

### CODE REVIEW

File: src/components/Guilds/AddressButton/index.tsx [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+const AddressButton: React.FC<AddressButtonProps> = ({
+  address,
+  chainId = DEFAULT_ETH_CHAIN_ID,
+  transactionsCounter,
+  onClick,
+}) => {
+  const { ensName, imageUrl, avatarUri } = useENSAvatar(address, chainId);
+
+  let imageUrlToUse = useMemo(() => {
```
Comment (hhamud) [inline src/components/Guilds/AddressButton/index.tsx]: use const here instead

File: src/hooks/Guilds/proposals/useProposal.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+import { mapStructToProposal } from '../../../utils/guildsProposals';
+import { useERC20Guild } from '../contracts/useContract';
+
+export interface useProposalReturns {
+  proposal: Proposal;
+  error: null | Error;
+  loading: boolean;
+}
+
+export const useProposal = (
```
Comment (hhamud) [inline src/hooks/Guilds/proposals/useProposal.ts:12]: same thing here.

File: src/hooks/Guilds/ipfs/useIPFSNode.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
+import { useEffect, useState } from 'react';
+import * as IPFS from 'ipfs-core';
+import CID from 'cids';
+
+export default function useIPFSNode() {
```
Comment (hhamud) [inline src/hooks/Guilds/ipfs/useIPFSNode.ts:5]: convert to es6 style.

File: src/hooks/Guilds/ipfs/useIPFSNode.ts [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    return cid.toString();
+  }
+
+  async function pin(hash: string) {
+    if (!isReady) return null;
+
+    const cid = new CID(hash);
+    console.debug('[useIPFSNode] Pinning IPFS CID', cid);
+    return ipfs.pin.add(cid.toString());
+  }
```
Comment (hhamud) [inline src/hooks/Guilds/ipfs/useIPFSNode.ts:40]: same thing here

**Review by hhamud — CHANGES_REQUESTED** [review]
Overall it works great, just a few stylistic changes.

**Comment by hhamud** [comment]
image
an issue here with the address icon

**Comment by rossneilson** [comment]
Think this is supposed to be below the title but above the description
Screenshot 2022-01-11 at 10 52 57

**Review by luchux — DISMISSED** [review]

**Review by hhamud — APPROVED** [review]
LGTM!

**Review by rossneilson — APPROVED** [review]

**Pull request merged**